perfkitbenchmarker/packages/ycsb.py

```diff
@@ -47,3 +143,533 @@ def YumInstall(vm):
+      options.
+
+    Returns:
+      List of sample.Sample objects.
+    """
+    all_results = []
+    for workload_index, workload_file in enumerate(workloads):
+      parameters = {'operationcount': FLAGS.ycsb_operation_count,
+                    'recordcount': FLAGS.ycsb_record_count}
+      if FLAGS.ycsb_timelimit:
+        parameters['timelimit'] = FLAGS.ycsb_timelimit
+      parameters.update(kwargs)
+      remote_path = os.path.join(vm_util.VM_TMP_DIR,
+                                 os.path.basename(workload_file))
+
+      with open(workload_file) as fp:
+        workload_meta = _ParseWorkload(fp.read())
+        workload_meta.update(kwargs)
+        workload_meta.update(workload_name=os.path.basename(workload_file),
+                             workload_index=workload_index,
+                             stage='run')
+
+      def PushWorkload(vm):
+        vm.PushFile(workload_file, remote_path)
+      vm_util.RunThreaded(PushWorkload, vms)
+
+      parameters['parameter_files'] = [remote_path]
+      for client_count in _GetThreadsPerLoaderList():
+        parameters['threads'] = client_count
+        results = self._RunThreaded(vms, **parameters)
+        client_meta = workload_meta.copy()
+        client_meta.update(clients=len(vms) * client_count,
+                           threads_per_client_vm=client_count)
+
+        if FLAGS.ycsb_include_individual_results and len(results) > 1:
+          for i, result in enumerate(results):
+            all_results.extend(_CreateSamples(
+                result,
+                result_type='individual',
+                result_index=i,
+                include_histogram=FLAGS.ycsb_histogram,
+                **client_meta))
+
+        combined = _CombineResults(results)
+        all_results.extend(_CreateSamples(
+            combined, result_type='combined',
+            include_histogram=FLAGS.ycsb_histogram,
+            **client_meta))
+
+    return all_results
+
+  def LoadAndRun(self, vms, workloads=None, load_kwargs=None, run_kwargs=None):
+    """Load data using YCSB, then run each workload/client count combination.
+
+    Loads data using the workload defined by 'workloads', then
+    executes YCSB for each workload file in 'workloads', for each
+    client count defined in FLAGS.ycsb_threads_per_client.
+
+    Generally database benchmarks using YCSB should only need to call this
+    method.
+
+    Args:
+      vms: List of virtual machines. VMs to use to generate load.
+      workloads: List of strings. Workload files to use. If unspecified,
+        _GetWorkloadFileList() is used.
+      load_kwargs: dict. Additional arguments to pass to the load stage.
+      run_kwargs: dict. Additional arguments to pass to the run stage.
+    Returns:
+      List of sample.Sample objects.
+    """
+    workloads = workloads or _GetWorkloadFileList()
+    assert workloads, 'no workloads'
+    load_samples = list(self._LoadThreaded(vms, workloads[0],
```

Comment:
Should we provide the option to skip reporting load_samples if people only care about the run samples?